pkg/operator/admission/webhook.go

```diff
@@ -0,0 +1,203 @@
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/csi"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"os"
+)
+
+const (
+	appName            = "rook-admission-controller"
+	secretVolumeName   = "webhook-certs"
+	secretMountPath    = "/etc/webhook"
+	serviceAccountName = "rook-admission-controller-sa"
+	storageFlag        = "ceph"
+)
+
+var (
+	validatePath       = "/validate"
+	servicePort  int32 = 443
+	serverPort   int32 = 8079
+	namespace          = os.Getenv(k8sutil.PodNamespaceEnvVar)
+)
+
+func isSecretPresent(context *clusterd.Context) (bool, error) {
+	_, err := context.Clientset.CoreV1().Secrets(namespace).Get(appName, metav1.GetOptions{})
+	if err != nil {
+		// If secret is not found. All good ! Proceed with rook without admission controllers
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func createWebhookService(context *clusterd.Context) error {
+	webhookService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: servicePort,
+					TargetPort: intstr.IntOrString{
+						IntVal: serverPort,
+					},
+				},
+			},
+
+			Selector: map[string]string{
+				"app": "rook-ceph-admission-controller",
+			},
+		},
+	}
+	_, err := context.Clientset.CoreV1().Services(namespace).Create(&webhookService)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// InitWebhookIfSecretPresent will initialize the webhook if secret is detected
+func InitWebhookIfSecretPresent(context *clusterd.Context, admissionImage string) error {
+	isPresent, err := isSecretPresent(context)
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve secret")
+	}
+	if isPresent {
+		err = initWebhook(context, admissionImage)
+		if err != nil {
+			return errors.Wrap(err, "failed to initialize webhook")
+		}
+	}
+	return nil
+}
+
+func initWebhook(context *clusterd.Context, admissionImage string) error {
+	// At this point volume should be mounted, so proceed with creating the service and validatingwebhookconfig
+	err := createWebhookService(context)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service")
+	}
+	err = createWebhookPods(context, admissionImage)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service")
+	}
+	return nil
+}
+
+func createWebhookPods(context *clusterd.Context, admissionImage string) error {
+	admission_parameters := []string{"admission-controller",
+		"--storage", storageFlag}
+	secretVolume := getSecretVolume()
+	secretVolumeMount := getSecretVolumeMount()
+
+	antiAffinity := csi.GetPodAntiAffinity("app", "rook-ceph-admission-controller")
```

Comment:
Define this const at the top of the source file. 
```
  appName = "rook-ceph-admission-controller"
```

```suggestion
	antiAffinity := csi.GetPodAntiAffinity(k8sutil.AppAttr, appName)
```
There are several places that you can make these changes to use `k8sutil.AppAttr` and `appName`.